quickwit/quickwit-proto/src/search/mod.rs

```diff
@@ -118,6 +166,67 @@ impl PartialOrd for SortValue {
     }
 }
 
+impl std::hash::Hash for SortValue {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        let this = self.normalize();
+        match this {
+            SortValue::U64(number) => {
+                0u8.hash(state);
+                number.hash(state);
+            }
+            SortValue::I64(number) => {
+                1u8.hash(state);
+                number.hash(state);
+            }
+            SortValue::F64(number) => {
+                2u8.hash(state);
+                number.to_bits().hash(state);
+            }
+            SortValue::Boolean(b) => {
+                3u8.hash(state);
+                b.hash(state);
+            }
+        }
+    }
+}
+
+impl SortValue {
+    pub fn normalize(&self) -> Self {
+        match self {
+            SortValue::U64(_) => *self,
+            SortValue::Boolean(_) => *self,
+            SortValue::I64(number) => {
+                if let Ok(number) = (*number).try_into() {
+                    SortValue::U64(number)
+                } else {
+                    *self
+                }
+            }
+            SortValue::F64(number) => {
+                let number = *number;
+                if number.ceil() == number {
+                    // number is not NaN, and is a natural number
+                    if number.is_sign_positive() && number <= u64::MAX as f64 {
+                        // number is positive (including positive zero) and less than u64::MAX, we
+                        // can convert it safely
+                        return SortValue::U64(number as u64);
+                    } else if number.is_sign_negative() && number >= i64::MIN as f64 {
```

Comment:
this goes against what we do outside in tantivy.

We tend to prefer i64 whenever possible.
so if ceil == number {

   if number within i64 range -> cast to i64 
   if number wthin. u64 range -> cast to u64
   else { do nothing }